types/evidence.go

```diff
@@ -213,42 +352,60 @@ func (l *LightClientAttackEvidence) ValidateBasic() error {
 	return nil
 }
 
-// String returns a string representation of LightClientAttackEvidence
-func (l *LightClientAttackEvidence) String() string {
-	return fmt.Sprintf("LightClientAttackEvidence{ConflictingBlock: %v, CommonHeight: %d}",
-		l.ConflictingBlock.String(), l.CommonHeight)
-}
-
 // ToProto encodes LightClientAttackEvidence to protobuf
 func (l *LightClientAttackEvidence) ToProto() (*tmproto.LightClientAttackEvidence, error) {
 	conflictingBlock, err := l.ConflictingBlock.ToProto()
 	if err != nil {
 		return nil, err
 	}
 
+	byzVals := make([]tmproto.Validator, len(l.ByzantineValidators))
+	for idx, val := range l.ByzantineValidators {
+		valpb, err := val.ToProto()
+		if err != nil {
+			return nil, err
+		}
+		byzVals[idx] = *valpb
+	}
+
 	return &tmproto.LightClientAttackEvidence{
-		ConflictingBlock: conflictingBlock,
-		CommonHeight:     l.CommonHeight,
+		ConflictingBlock:    conflictingBlock,
+		CommonHeight:        l.CommonHeight,
+		ByzantineValidators: byzVals,
+		TotalVotingPower:    l.TotalVotingPower,
+		Timestamp:           l.Timestamp,
 	}, nil
 }
 
 // LightClientAttackEvidenceFromProto decodes protobuf
-func LightClientAttackEvidenceFromProto(l *tmproto.LightClientAttackEvidence) (*LightClientAttackEvidence, error) {
-	if l == nil {
+func LightClientAttackEvidenceFromProto(lpb *tmproto.LightClientAttackEvidence) (*LightClientAttackEvidence, error) {
+	if lpb == nil {
 		return nil, errors.New("empty light client attack evidence")
 	}
 
-	conflictingBlock, err := LightBlockFromProto(l.ConflictingBlock)
+	conflictingBlock, err := LightBlockFromProto(lpb.ConflictingBlock)
 	if err != nil {
 		return nil, err
 	}
 
-	le := &LightClientAttackEvidence{
-		ConflictingBlock: conflictingBlock,
-		CommonHeight:     l.CommonHeight,
+	byzVals := make([]Validator, len(lpb.ByzantineValidators))
+	for idx := range lpb.ByzantineValidators {
```

Comment:
```suggestion
	for idx, val := range lpb.ByzantineValidators {
```

Reply:
This change was made to fix some aliasing lint problem. I forget now what exactly

Reply:
Probably because you redefine `val`?